src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,502 @@
+                                    coordinateLocation, true);
+
+                            // There is at least one point at this location in the raw Atlas. Update
+                            // all existing points to have the country code.
+                            for (final Point rawAtlasPoint : rawAtlasPointsAtSliceVertex)
+                            {
+                                // Update the country codes
+                                this.afterSlicingLinesAndPoints
+                                        .updatePointTags(rawAtlasPoint.getIdentifier(), pointTags);
+
+                                // Add all point identifiers to make up the new Line
+                                newLineShapePoints.add(rawAtlasPoint.getIdentifier());
+                            }
+                        }
+                    }
+                }
+
+                // Extract relevant tag values for this slice
+                final Map<String, String> lineTags = createLineTags(slice);
+
+                // Create and store the new line
+                final TemporaryLine createdLine = new TemporaryLine(
+                        lineIdentifierFactory.nextIdentifier(), newLineShapePoints, lineTags);
+                createdLines.add(createdLine);
+            }
+
+            // Update the change with the added and removed lines
+            createdLines.forEach(
+                    createdLine -> this.afterSlicingLinesAndPoints.createLine(createdLine));
+
+            this.afterSlicingLinesAndPoints.deleteLine(line.getIdentifier());
+            this.afterSlicingLinesAndPoints.createDeletedToCreatedMapping(line.getIdentifier(),
+                    createdLines.stream().map(TemporaryLine::getIdentifier)
+                            .collect(Collectors.toList()));
+        }
+        else
+        {
+            // TODO - Consider expanding to a 4 digit namespace for identifiers
+            logger.error("Country slicing exceeded maximum identifier name space of {} for Line {}",
+                    AbstractIdentifierFactory.IDENTIFIER_SCALE, line.getIdentifier());
+        }
+    }
+
+    /**
+     * Converts the given {@link Line} to a JTS {@link Geometry}, slices the geometry and updates
+     * all corresponding {@link Point}s and {@link Line}s in the given raw Atlas.
+     *
+     * @param line
+     *            The {@link Line} to slice
+     */
+    private void sliceLine(final Line line)
+    {
+        this.statistics.recordProcessedLine();
+        final List<Geometry> slices = convertToJtsGeometryAndSlice(line);
+        processSlices(line, slices);
+    }
+
+    /**
+     * Slices all the {@link Line}s in the given raw Atlas.
+     */
+    private void sliceLines()
+    {
+        this.rawAtlas.lines().forEach(this::sliceLine);
+    }
+
+    /**
+     * Updates all stand-alone {@link Point}s that haven't been assigned a country code after
+     * way-slicing.
```

Comment:
Maybe expand the comment here to explain that it means they fell outside of any country boundary?

Reply:
Good point